Please add support for Python 3.13

Hello, 

I just add a bug report in order to trace the Python 3.13 support.

Python 3.13 is already part of Debian and will be the next default interpreter of trixie (Debian 13)

Here a problem observed during the tests

```
removing build/bdist.linux-x86_64/wheel
Successfully built hyperspy-0.0.0-py3-none-any.whl
I: pybuild plugin_pyproject:144: Unpacking wheel built for python3.12 with "installer" module
   dh_auto_test -O--buildsystem=pybuild
I: pybuild base:311: cd /<<PKGBUILDDIR>>/.pybuild/cpython3_3.13_hyperspy/build; python3.13 -m pytest --ignore hyperspy/tests/drawing/test_plot_markers.py --ignore hyperspy/tests/drawing/test_plot_signal1d.py --ignore hyperspy/tests/drawing/test_plot_signal2d.py
============================= test session starts ==============================
platform linux -- Python 3.13.0, pytest-8.3.3, pluggy-1.5.0
Matplotlib: 3.8.3
Freetype: 2.13.3
rootdir: /<<PKGBUILDDIR>>/.pybuild/cpython3_3.13_hyperspy/build
configfile: pyproject.toml
testpaths: hyperspy
plugins: instafail-0.5.0, typeguard-4.4.1, xdist-3.6.1, mpl-0.17.0, rerunfailures-14.0
created: 64/64 workers
64 workers [3191 items]

................................EEE..EEEE.....E....E.........EE......EF. [  2%]
.E.EF.........EE..F.........F..EEEE......F...E..FEF.....F..E..FE..EEFF.. [  4%]
EE..EEF.FE...EEE.....FEEE.EEEE...EEE.EFEEEEEEEEEFF...FEFFF..FFFEEF.EEEEF [  6%]
...EEEEF.....EEEFF......ss..s.....EEEEFEEEEEFE.E.EFEFEEFF.EEFEEEFEEE...F [  9%]
FEFFEFEE..EEEFEEFEFEEFEEE..FEFFFEEEEEEEEFF....EEEEEFEEEFEEFFE.EEEEE.EEFE [ 11%]
EEEEEFFEE.EEE.EFEFEFFEFEFE.EFE.FEEEEEEFEEEFEF.EF.EEEEFEFFFEE.FEEEE.EEEEE [ 13%]
EEEFE.E.EEFEE.EEEEEEFEEEE.E.FFEE.F.FE.E.EE.EF.F.EE.E.EE..FE.FE.EEE..E..E [ 15%]
FEE...FEEEEFE....EFEE...E.EE..E.EFE.EEEFEEFE...EE..EEE.EEEEF....EEF..F.E [ 18%]
.FEEE.FEEFEEEFEEEFEEEEEFEEEFEEEEFFEEEEE..EEEEEEEEFFEEFEEEEEEFFEFEEEEEEFE [ 20%]
FFEEEEEFEEEEEEFEEEEF.EFEFEEEFEEEEFEEEEEEEEEEFEFEEEEEEFEFEEFFEFEEEEFFEEEE [ 22%]
FEEFFEFF.EEEEEEEFEEEFEEEEEEEEEFEEE.EEFEFEFEEEEFEEEFEFEEFEFFFEFEFEEEFEEEF [ 24%]
FFEEEEEEEFE.EEEEEEE.FEEFFEEEEEFEFEFFEFEFEEFFEEEFEEEFFEEEEFFEEFFEFFEEEEEE [ 27%]
FFFEEEEEEEEEEEEFFFEEEFEEFEEEEFEEFEEEEFEEEEFEEFFEEEEFEFEF.FFEFFEE.EFFFEEF [ 29%]
EFFEEEFFEEEFFFEEEEFFFFEFFFEEEEFEEEEFEFEEEEEFEEEFFFEEEFEEEEFEFFFEEFFEFFEE [ 31%]
EFFEFFFFFEEFEEEEEEEFFFFFF.FsEEEFEEFFE.EFFFEEEEEEEFFEEEEEFEFFFEEEFFFFEEEF [ 33%]
FEEEFFEFFFEFEEFFFEEEFEEFEEEFFEFFFEFEFFsEEEEFEFEEEEEEEFEEEFEFEFFsEsFFFEEF [ 36%]
Ex.FEEEEEEEFFFEFFEEEFEFFEEFEEFEFFFEFFFEFFFE.EEFEEEEEEEEFFEEEEFEEFEF.EEEE [ 38%]
EFFFEEEFE...E.FE.FF..F.FE.FFFEFEEFEFFEFFEEFExEEFFEEEEFEFEEEFEFEFFEEFEEFF [ 40%]
EEEEEEEEFEEFFEEFFFEFEEFFEFFFEFFEEFEEFEEFEEEEFEEEFEEEEFFEEEEEEFEF.EF.FEEE [ 42%]
EEEEFEEFEEEFEEEFFFFFEFEFEEFEEFFEEEEEEEEFEEEFEFEFEFFEEEFEEEEEEEEFEEEF.E.. [ 45%]
EFEFF..EEF.FEEEEE.FFFFEEFEEEEEEEFEEEEEEFEEFFFEEFEEFFEEEEEEEFEFEFEEEFEEEE [ 47%]
EEFEFEEEFEEEEFEEEFEEEEEFEEFEEEE.EEEFFEEFEFEEEFEFEEFFEFEEEFFEEEFEEEFEEEEF [ 49%]
F...FFEE.E.FEEEFEEFFFEEEEEEEsFEEEFFFE.FEFEFE.E.EEEEFFE.EFEEFEFEEFFEFEEEF [ 51%]
.EEEEFEEEEF.FFEEEE.....E..F..FFFEEEFFEFFEEEEEFEEEFFFEEFEEEEEFEEEEEEF.FEE [ 54%]
FFEEFFE.E.E..E.EEFFFFEEEFFFEFEFFFEFEFFEEEFEEEFEEEEEEFEFFEEEEFEFEEFFEFEFE [ 56%]
E.EE.E.E.FE.EF.EE.EFFFEFFFEEFFFEFEEFE.FEFFEs.EEFE..EEEEFEFEFEFEEEEEEEFE. [ 58%]
E.E.EEEFFFFEFFF.F.E.EEEEFFFFEFEEEFFFFF.E..EEEFEEEEF.EF..FEEEFsEEEEFEFFE. [ 60%]
EEFFFE.FEFFEEFEEEEFEFEFEEEEEFFFEEEEFFE.EFEEEEFFEFEFEEEEFFFEFFEFEFEFFEFFE [ 63%]
EFEFEEFEEFFEFFEEEEEFFEEFFEEEEEFEFFEEEFFEEEEEFEEFEEFEEFFEEFEEEEFEEEEEFFEE [ 65%]
EEEFEEEFEFEFFFEEEEEFEFEFFEEEEEFEFEFEEFEFFEEEEFFEEEEEFFEEEEEEEFFEEFEEEFFF [ 67%]
EEFEEFEEEEEEFFEEFEFEEEFEEFFEE.EEEEEFEFFFEEEFEFFEE.EEEFEFEEEEFEFFEFEFEFEE [ 69%]
EEEEEEEE.EFEEFEEFEEFEFEE.EEEEEEEEEFFEFEEEEF.EEEEEEEEEFEEFFEEEEEEEEEEEEE. [ 72%]
E.FEFEFEEEEEEEFEEEEFFFEEFEEEEEEEEFEFEEEFFEEFEEEEEEEF.EEEEFFEFEEEEEFEEEEE [ 74%]
FFFEEEEEEEFEEEEFFEEEFFEEEEEEEEEEFEFEEEEEEEEEFEEFEEEEEEEEE.EFEEEFEEEEEEEE [ 76%]
FEEEEFEEEEEEEEFEEEEFEFEEEEEFEEEEEEEFFEEEEEFEEEEFEEEEEEEEEEEFEFEEFEEEEEEE [ 78%]
FEEEFFEEEEEEEFEEFEEEEFEEEEEEEFEEEFEFEEEEEEFEFEEFEEEEEEEFEFEFEEEFEEEEEFEE [ 81%]
FFEEEEEEFFEEEE.FE.EEEEEFEEEEEFEEEEFEEEEFEEEEEEEFFEEEEEEEFEEFEEEEEEEEEEEF [ 83%]
FEEEEFEEEEEFFFEEEEEEEEEEFEEFEEEEEEEEEFEEFEEEEEEEEEFEFEEEEEEEFEEEEFEEEEEE [ 85%]
EFEEEEEFEFEEEEFEEEEFEEEFEFEEEEEFEEEFFEEEFEEEFEFEEEEFEEFEEFEE.EEFEEEFFEEE [ 87%]
EEEFEEFFEEEEEFFEEEEEEEEFEFEEEEEEFEEFEEEFEFEEEFEFEEFEEEEFFEEEEFEEFFEEFEEE [ 90%]
EEFFEEFE.EEEFEFFFFEEEEFEFFFEFEEEE.FFFE.EEEFF.EFEEE.FF.EEFEFFE.EE.EFFE.EE [ 92%]
.FEFE.EEFE.FEEEFE.FEEEF.EEFEEFE.EEEFFEEFEEFEEEFEE.F..FEEFEFFEFEEFEFEEFEF [ 94%]
EFEEFEEEEFEEEEFEEFEEFEEFEEEFEEFEEEEEEEEEEEEEEEEEEEEEEEEFEEFEEFEEEEFEEEFE [ 97%]
EFEEFEEFEEEEEFEEFEEFEEFEEFEEFEEEFEEFEFEsEFEFEFEFFFFFFFFEEEEEEEEEEEFFFFFF [ 99%]
FFFFFFFFFFFFEEEEFFFFFFF                                                  [100%]
==================================== ERRORS ====================================
___________ ERROR collecting hyperspy/tests/drawing/test_markers.py ____________
hyperspy/tests/drawing/test_markers.py:71: in <module>
    class TestMarkers:
hyperspy/tests/drawing/test_markers.py:195: in TestMarkers
    UniformDataAxis(scale=0.5, offset=-1),
hyperspy/axes.py:1239: in __init__
    super().__init__(
hyperspy/axes.py:300: in __init__
    self.events.index_changed = Event(
hyperspy/events.py:186: in __init__
    self._trigger_maker(arguments)
hyperspy/events.py:228: in _trigger_maker
    new_f = locals()["trigger"]
E   KeyError: 'trigger'
___________ ERROR collecting hyperspy/tests/drawing/test_markers.py ____________
```

I do not have these issue with Python 3.12

```
I: pybuild base:311: cd /<<PKGBUILDDIR>>/.pybuild/cpython3_3.12_hyperspy/build; python3.12 -m pytest --ignore hyperspy/tests/drawing/test_plot_markers.py --ignore hyperspy/tests/drawing/test_plot_signal1d.py --ignore hyperspy/tests/drawing/test_plot_signal2d.py
============================= test session starts ==============================
platform linux -- Python 3.12.7, pytest-8.3.3, pluggy-1.5.0
Matplotlib: 3.8.3
Freetype: 2.13.3
rootdir: /<<PKGBUILDDIR>>/.pybuild/cpython3_3.12_hyperspy/build
configfile: pyproject.toml
testpaths: hyperspy
plugins: instafail-0.5.0, typeguard-4.4.1, xdist-3.6.1, mpl-0.17.0, rerunfailures-14.0
created: 64/64 workers
64 workers [3897 items]

........................................................................ [  1%]
........................................................................ [  3%]
........................................................................ [  5%]
........................................................................ [  7%]
...................ss..s........................EEE..................... [  9%]
.......................................................s................ [ 11%]
....EEE.....................s...........s.......E........Es......E...... [ 12%]
..........E...........E........E......................s.s............... [ 14%]
......................EEEEEE.........................E...EE............. [ 16%]
...E.............E..............E.........E.........E...............E... [ 18%]
........E..E..............E......E.......E.............................. [ 20%]
.........s.............................................................. [ 22%]
........................................................................ [ 24%]
.........EF............................................................. [ 25%]
....E................................................................... [ 27%]
...........E............................................................ [ 29%]
..............................................................E......... [ 31%]
...............................................................F........ [ 33%]
...............................E............E........................... [ 35%]
............................................................F..........F [ 36%]
....................................................................F... [ 38%]
.....................................................F.........F...F.... [ 40%]
.....F......F..................F................F....................... [ 42%]
.................F.........E........E..........E.........E......F....... [ 44%]
..........F.....s.......s...................................F........... [ 46%]
.........F.....................................F..........s............. [ 48%]
........................................................................ [ 49%]
................F....................................F...............sF. [ 51%]
...ss...................................................F............... [ 53%]
....................s....s........................s..................... [ 55%]
..................F..................................................... [ 57%]
............................E..s........................................ [ 59%]
........................................................................ [ 60%]
........................................................................ [ 62%]
........................................................................ [ 64%]
............F..........F................................................ [ 66%]
........................................................................ [ 68%]
........................................................................ [ 70%]
.............................................................F.......... [ 72%]
....................................................................s..s [ 73%]
.s......................x............................................... [ 75%]
........................................................................ [ 77%]
........................................................................ [ 79%]
...................................................s.................... [ 81%]
........................................................................ [ 83%]
.............s................s.s.................x..................... [ 84%]
........................................................................ [ 86%]
........................................................................ [ 88%]
........................................................................ [ 90%]
........................................................................ [ 92%]
........................................................................ [ 94%]
........................................................................ [ 96%]
........................................................................ [ 97%]
..ssssssss.F............................................................ [ 99%]
..........                                                               [100%]

____________ ERROR at setup of TestSubPixelAlign.test_align_subpix _____________
[gw18] linux -- Python 3.12.7 /usr/bin/python3.12

self = <hyperspy.tests.signals.test_2D_tools.TestSubPixelAlign object at 0x7f12ac2d9340>
method = <bound method TestSubPixelAlign.test_align_subpix of <hyperspy.tests.signals.test_2D_tools.TestSubPixelAlign object at 0x7f12ac2d9340>>

    def setup_method(self, method):
>       ref_image = ascent()

hyperspy/tests/signals/test_2D_tools.py:49: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3/dist-packages/scipy/datasets/_fetchers.py:71: in ascent
    fname = fetch_data("ascent.dat")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

dataset_name = 'ascent.dat', data_fetcher = None

    def fetch_data(dataset_name, data_fetcher=data_fetcher):
        if data_fetcher is None:
>           raise ImportError("Missing optional dependency 'pooch' required "
                              "for scipy.datasets module. Please use pip or "
                              "conda to install 'pooch'.")
E           ImportError: Missing optional dependency 'pooch' required for scipy.datasets module. Please use pip or conda to install 'pooch'.

/usr/lib/python3/dist-packages/scipy/datasets/_fetchers.py:27: ImportError
------------------------------ Captured log setup ------------------------------
WARNING  matplotlib.testing:__init__.py:38 Could not set locale to English/United States. Some date-related tests may fail.
```

here the error are related to the network access whcih is not allowed  during the build

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -84,18 +84,6 @@ jobs:
           python --version
           pip --version
 
-      - name: Install traits dev for python 3.13
-        if: ${{ matrix.PYTHON_VERSION == '3.13' }}
-        run: |
-          # traits release with python 3.13 pending
-          pip install git+https://github.com/enthought/traits.git
-
-      - name: Install scikit-image 0.25.0-rc1
-        if: ${{ matrix.PYTHON_VERSION == '3.13' }}
-        run: |
-          # speed up build until scikit-image 0.25 is released
-          pip install scikit-image --pre
-
       - name: Install
         shell: bash
         run: |

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -16,7 +16,7 @@ jobs:
       matrix:
         os: [ubuntu, windows, macos]
         os_version: [latest]
-        PYTHON_VERSION: ['3.10', '3.11', ]
+        PYTHON_VERSION: ['3.10', '3.12']
         PIP_SELECTOR: ['[all, tests, coverage]']
         include:
           # test oldest supported version of main dependencies on python 3.8
@@ -27,7 +27,7 @@ jobs:
             OLDEST_SUPPORTED_VERSION: true
             # Don't install pillow 10.4.0 because of its incompatibility with numpy 1.20.x
             # https://github.com/python-pillow/Pillow/pull/8187
-            DEPENDENCIES: dask[array]==2022.9.2 matplotlib==3.6.0 numba==0.53 numpy==1.20.0 scipy==1.6 scikit-image==0.18 scikit-learn==1.0.1 pillow!=10.4.0
+            DEPENDENCIES: dask[array]==2022.9.2 matplotlib==3.6.0 numba==0.53 numpy==1.20.0 scipy==1.6 sympy==1.10 scikit-image==0.18 scikit-learn==1.0.1 pillow!=10.4.0 numexpr==2.8
             LABEL: -oldest
           # test minimum requirement
           - os: ubuntu
@@ -41,11 +41,11 @@ jobs:
             PIP_SELECTOR: '[all, tests, coverage]'
           - os: ubuntu
             os_version: latest
-            PYTHON_VERSION: '3.12'
+            PYTHON_VERSION: '3.11'
             PIP_SELECTOR: '[all, tests, coverage]'
           - os: macos
             os_version: '13'
-            PYTHON_VERSION: '3.12'
+            PYTHON_VERSION: '3.11'
             PIP_SELECTOR: '[all, tests, coverage]'
 
     steps:

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -16,36 +16,36 @@ jobs:
       matrix:
         os: [ubuntu, windows, macos]
         os_version: [latest]
-        PYTHON_VERSION: ['3.9', '3.10', ]
+        PYTHON_VERSION: ['3.10', '3.11', ]
         PIP_SELECTOR: ['[all, tests, coverage]']
         include:
           # test oldest supported version of main dependencies on python 3.8
           - os: ubuntu
             os_version: latest
-            PYTHON_VERSION: '3.8'
+            PYTHON_VERSION: '3.9'
             PIP_SELECTOR: '[all, tests, coverage]'
             OLDEST_SUPPORTED_VERSION: true
             # Don't install pillow 10.4.0 because of its incompatibility with numpy 1.20.x
             # https://github.com/python-pillow/Pillow/pull/8187
-            DEPENDENCIES: dask[array]==2022.9.2 matplotlib==3.6.0 numba==0.52 numpy==1.20.0 scipy==1.6 scikit-image==0.18 scikit-learn==1.0.1 pillow!=10.4.0
+            DEPENDENCIES: dask[array]==2022.9.2 matplotlib==3.6.0 numba==0.53 numpy==1.20.0 scipy==1.6 scikit-image==0.18 scikit-learn==1.0.1 pillow!=10.4.0
             LABEL: -oldest
           # test minimum requirement
           - os: ubuntu
             os_version: latest
-            PYTHON_VERSION: '3.12'
+            PYTHON_VERSION: '3.13'
             PIP_SELECTOR: '[tests, coverage]'
             LABEL: -minimum
           - os: ubuntu
             os_version: latest
-            PYTHON_VERSION: '3.8'
+            PYTHON_VERSION: '3.9'
             PIP_SELECTOR: '[all, tests, coverage]'
           - os: ubuntu
             os_version: latest
-            PYTHON_VERSION: '3.11'
+            PYTHON_VERSION: '3.12'
             PIP_SELECTOR: '[all, tests, coverage]'
           - os: macos
             os_version: '13'
-            PYTHON_VERSION: '3.11'
+            PYTHON_VERSION: '3.12'
             PIP_SELECTOR: '[all, tests, coverage]'
 
     steps:
@@ -84,15 +84,27 @@ jobs:
           python --version
           pip --version
 
-      - name: Install oldest supported version
-        if: ${{ matrix.OLDEST_SUPPORTED_VERSION }}
+      - name: Install traits dev for python 3.13
+        if: ${{ matrix.PYTHON_VERSION == '3.13' }}
         run: |
-          pip install ${{ matrix.DEPENDENCIES }} -v
+          # traits release with python 3.13 pending
+          pip install git+https://github.com/enthought/traits.git
+
+      - name: Install scikit-image 0.25.0-rc1
+        if: ${{ matrix.PYTHON_VERSION == '3.13' }}
+        run: |
+          # speed up build until scikit-image 0.25 is released
+          pip install scikit-image --pre
 
       - name: Install
         shell: bash
         run: |
           pip install ${{ env.PIP_ARGS }} .'${{ matrix.PIP_SELECTOR }}'
+
+      - name: Install oldest supported version
+        if: ${{ matrix.OLDEST_SUPPORTED_VERSION }}
+        run: |
+          pip install ${{ matrix.DEPENDENCIES }} -v
 
       - name: Pip list
         run: |

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -27,27 +27,27 @@ strategy:
   matrix:
     Linux_Python39:
       vmImage: 'ubuntu-latest'
-      PYTHON_VERSION: '3.9'
+      PYTHON_VERSION: '3.10'
       MINIFORGE_PATH: $(Agent.BuildDirectory)/miniforge3
     Linux_Python310:
       vmImage: 'ubuntu-latest'
-      PYTHON_VERSION: '3.10'
+      PYTHON_VERSION: '3.11'
       MINIFORGE_PATH: $(Agent.BuildDirectory)/miniforge3
     MacOS_Python39:
       vmImage: 'macOS-latest'
-      PYTHON_VERSION: '3.9'
+      PYTHON_VERSION: '3.10'
       MINIFORGE_PATH: $(Agent.BuildDirectory)/miniforge3
     MacOS_Python310:
       vmImage: 'macOS-latest'
-      PYTHON_VERSION: '3.10'
+      PYTHON_VERSION: '3.11'
       MINIFORGE_PATH: $(Agent.BuildDirectory)/miniforge3
     Windows_Python39:
       vmImage: 'windows-latest'
-      PYTHON_VERSION: '3.9'
+      PYTHON_VERSION: '3.10'
       MINIFORGE_PATH: $(Agent.BuildDirectory)\miniforge3
     Windows_Python310:
       vmImage: 'windows-latest'
-      PYTHON_VERSION: '3.10'
+      PYTHON_VERSION: '3.11'
       MINIFORGE_PATH: $(Agent.BuildDirectory)\miniforge3
 
 pool:

--- a/conda_environment.yml
+++ b/conda_environment.yml
@@ -10,7 +10,7 @@ dependencies:
 - matplotlib-base >=3.6
 - natsort
 - numba >=0.53
-- numexpr
+- numexpr >=2.8
 - numpy >=1.20
 - packaging
 - pint >=0.10
@@ -22,6 +22,6 @@ dependencies:
 - scikit-image >=0.18
 - scikit-learn >=1.0.1
 - scipy >=1.6.0
-- sympy >=1.6.0
+- sympy >=1.10
 - tqdm >=4.9.0
 - traits >=4.5.0

--- a/conda_environment.yml
+++ b/conda_environment.yml
@@ -9,7 +9,7 @@ dependencies:
 - jinja2
 - matplotlib-base >=3.6
 - natsort
-- numba >=0.52
+- numba >=0.53
 - numexpr
 - numpy >=1.20
 - packaging

--- a/hyperspy/events.py
+++ b/hyperspy/events.py
@@ -18,6 +18,7 @@
 
 import inspect
 import re
+import sys
 from collections.abc import Iterable
 from contextlib import contextmanager
 from functools import wraps  # noqa: F401 Used in exec statement
@@ -224,7 +225,12 @@ class Event:
         gl = dict(globals())
         gl.update(locals())
         gl.update({"f": orig_f})  # Make sure it keeps the original!
-        exec(wrap_code, gl, locals())
+
+        if sys.version_info.minor >= 13:
+            locals_ = sys._getframe().f_locals
+        else:
+            locals_ = locals()
+        exec(wrap_code, gl, locals_)
         new_f = locals()["trigger"]
         # Replace the trigger function with the new one
         if defaults:

--- a/hyperspy/io.py
+++ b/hyperspy/io.py
@@ -456,7 +456,8 @@ def load(
     elif isgenerator(filenames):
         filenames = list(filenames)
 
-    elif not isinstance(filenames, (list, tuple, MutableMapping)):
+    # pathlib.Path.glob returns a map object in python 3.13
+    elif not isinstance(filenames, (list, tuple, MutableMapping, map)):
         raise ValueError(
             "The filenames parameter must be a list, tuple, "
             f"string or None, not {type(filenames)}"

--- a/hyperspy/tests/drawing/test_plot_signal1d.py
+++ b/hyperspy/tests/drawing/test_plot_signal1d.py
@@ -74,14 +74,15 @@ def _matplotlib_pick_event(figure, click, artist):
 
 @pytest.fixture
 def setup_teardown(request, scope="class"):
-    plot_testing = request.config.getoption("--mpl")
     pytest_mpl_spec = importlib.util.find_spec("pytest_mpl")
 
     if pytest_mpl_spec is None:
         mpl_generate_path_cmdopt = None
+        plot_testing = False
     else:
         # This option is available only when pytest-mpl is installed
         mpl_generate_path_cmdopt = request.config.getoption("--mpl-generate-path")
+        plot_testing = request.config.getoption("--mpl")
 
     # SETUP
     # duplicate baseline images to match the test_name when the

--- a/hyperspy/tests/signals/test_cuda.py
+++ b/hyperspy/tests/signals/test_cuda.py
@@ -21,6 +21,7 @@ import dask
 import numpy as np
 import pytest
 
+import hyperspy
 import hyperspy.api as hs
 
 try:
@@ -154,7 +155,7 @@ def test_to_device(lazy):
     s = hs.signals.Signal1D(data)
     if lazy:
         s = s.as_lazy()
-        assert isinstance(s, hs.hyperspy._signals.signal1d.LazySignal1D)
+        assert isinstance(s, hyperspy._signals.signal1d.LazySignal1D)
         with pytest.raises(BaseException):
             s.to_device()
     else:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -11,11 +11,11 @@ classifiers = [
     "Intended Audience :: Science/Research",
     "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
     "Operating System :: OS Independent",
-    "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
     "Topic :: Scientific/Engineering",
     "Topic :: Scientific/Engineering :: Physics",
     "Topic :: Software Development :: Libraries",
@@ -83,7 +83,7 @@ keywords = [
 ]
 name = "hyperspy"
 readme = "README.md"
-requires-python = ">=3.8"
+requires-python = ">=3.9"
 
 [project.license]
 file = "COPYING.txt"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -38,7 +38,7 @@ dependencies = [
     "rosettasciio[hdf5]",
     "scikit-image>=0.18",
     "scipy>=1.6.0",
-    "sympy>=1.6",
+    "sympy>=1.10",
     "tqdm>=4.9.0",
     "traits>=4.5.0",
 ]
@@ -141,7 +141,7 @@ learning = [
 ]
 speed = [
     "numba>=0.53",
-    "numexpr",
+    "numexpr>=2.8",
 ]
 tests = [
     "pytest-instafail",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -140,7 +140,7 @@ learning = [
     "scikit-learn>=1.0.1",
 ]
 speed = [
-    "numba",
+    "numba>=0.53",
     "numexpr",
 ]
 tests = [

--- a/upcoming_changes/3468.maintenance.rst
+++ b/upcoming_changes/3468.maintenance.rst
@@ -1,0 +1,1 @@
+Add support for python 3.13.

